controllers/serviceaccount_controller.go

```diff
@@ -254,9 +261,7 @@ func (r *ServiceAccountReconciler) ensureProviderServiceAccounts(ctx context.Con
 
 	for i, pSvcAccount := range pSvcAccounts {
 		// Note: We have to use := here to not overwrite log & ctx outside the for loop.
```

Comment:
Now that the request is properly added as a "VSphereCluster" k/v pair and not as "ProviderServiceAccount" we can simply set "ProviderServiceAccount" as a k/v pair here